### PR TITLE
various rd issues

### LINKED ISSUE #1114: BUG: Recurring Donations start off with deleted Opps, which then seem to get recreated.
full details, video, and steps in the hub: https://powerofus.force.com/0D58000001YDhPV

### LINKED ISSUE #1264: BUG: When using the recurring donation "Resfresh Opportunity" buttons, newly created opps disappear
After clicking the recurring donation "Refresh Opportunity" button, newly created opportunity in the related list have been deleted. User needs to refresh the page for a link to the newly created record. Seems like our code may be deleting and recreating opportunities unnecessarily or at the wrong time.

### LINKED ISSUE #1359: BUG: Recurring Donations with -none- for the Open Ended Status exhibit weird behavior.
Here is the original hub thread related to this: https://powerofus.force.com/0D58000001YDhPV
Here is the original bug we fixed related to this: bug #1114. https://github.com/SalesforceFoundation/Cumulus/issues/1114
Also related to bug #1246 https://github.com/SalesforceFoundation/Cumulus/issues/1264
Here are updated reproduceable instructions (from the post): 
Jeff Issenberg (Exponent Partners)
I've just run into this again so I did some more systematic testing. I seem to be getting this result when I choose "--None--" which is different than choosing "None" for the Open Ended Status. Do others get the same result? I've got Donations listed in the Recurring Donation's Donations related list. When I open any of these donations I get "Record deleted". Then 4 minutes later when I open the records they are there!! Strange.

### PR DESCRIPTION
**Pull request by davidhabib**
# Warning

# Info

# Issues
Fixes #1359 
Fixes #1264 
Fixes #1114

### CODE REVIEW

**Comment by davidhabib** [comment]
@ceiroa This pull request is ready for merge. thanks.

File: src/classes/RD_RecurringDonations.cls [hunk 1]
```diff
@@ -522,14 +624,19 @@ public with sharing class RD_RecurringDonations {
                 ERR_Handler.processErrors(errors.errorRecords, ERR_Handler_API.Context.RD.name());
             }            
             
-            if (!system.isBatch() && !system.isFuture() && !RD_ProcessControl.batchButton)
+            if (updateRecurringDonationSet.size() > 1 && !system.isBatch() && !system.isFuture() && !RD_ProcessControl.batchButton)
```
Comment (ceiroa) [inline src/classes/RD_RecurringDonations.cls]: Are the changes in lines 616 and 627 what fixes https://github.com/SalesforceFoundation/Cumulus/issues/1114?

File: src/classes/RD_RecurringDonations.cls [hunk 2]
```diff
@@ -920,7 +1057,7 @@ public with sharing class RD_RecurringDonations {
             }
                     
             if (ConIDForRole == null /* || Test.isRunningTest() */){
-                if (!system.isFuture() && !system.isBatch() && !RD_ProcessControl.batchButton)
+                if (createdOppIds.size() > 1 && !system.isFuture() && !system.isBatch() && !RD_ProcessControl.batchButton)
```
Comment (ceiroa) [inline src/classes/RD_RecurringDonations.cls]: Is this what fixes https://github.com/SalesforceFoundation/Cumulus/issues/1264?

**Comment by ceiroa** [comment]
The code changes are straightforward, and I can merge them any time. I'm just curious to know which changes fix which issues. I just saw the video related to #1114, and was quite surprised.

**Pull request merged**